core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -479,9 +479,21 @@ trait VecLike[T <: Data] extends collection.IndexedSeq[T] with HasId with Source
   * RTL writers should use [[Bundle]].  See [[Record#elements]] for an example.
   */
 abstract class Record(private[chisel3] implicit val compileOptions: CompileOptions) extends Aggregate {
+
+  // Doing this earlier than onModuleClose allows field names to be available for prefixing the names
+  // of hardware created when connecting to one of these elements
+  private def setElementRefs(): Unit = {
+    // Since elements is a map, it is impossible for two elements to have the same
+    // identifier; however, Namespace sanitizes identifiers to make them legal for Firrtl/Verilog
+    // which can cause collisions
+    val _namespace = Namespace.empty
+    for ((name, elt) <- elements) { elt.setRef(this, _namespace.name(name, leadingDigitOk=true)) }
```

Comment:
It looks like setRef doesn't have any protection against overwriting refs. Is there ever a case where overwriting a ref might be intentional, or does it make sense to add a guard - either here, or in setRef? Especially if there are now two call sites to setElementRefs (yes, in theory they should never both happen for any object - but defensive programming is generally a good practice)

Reply:
I don't think we ever want to override it so I think adding a guard makes sense

Reply:
Done